tests/timeseries/_formula_engine/test_formula_composition.py

```diff
@@ -171,6 +171,61 @@ async def test_formula_composition_missing_bat(self, mocker: MockerFixture) -> N
 
         assert count == 10
 
+    async def test_formula_composition_min_max(self, mocker: MockerFixture) -> None:
+        """Test the composition of formulas with min/max values."""
+        mockgrid = MockMicrogrid(grid_side_meter=True)
+        await mockgrid.start(mocker)
+
+        logical_meter = microgrid.logical_meter()
+        engine_min = logical_meter.grid_power.min(Power.from_watts(0)).build(
+            "grid_power_min"
+        )
+        engine_min_rx = engine_min.new_receiver()
+        engine_max = logical_meter.grid_power.max(Power.from_watts(0)).build(
+            "grid_power_max"
+        )
+        engine_max_rx = engine_max.new_receiver()
+
+        await mockgrid.mock_resampler.send_meter_power([100.0])
+
+        # Test min
+        min_pow = await engine_min_rx.receive()
+        assert (
+            min_pow is not None
+            and min_pow.value is not None
+            and math.isclose(min_pow.value.base_value, 0.0)
+        )
```

Comment:
Optionally you might want to define this assertion and similar ones in a more "pythonic" way
```suggestion
        assert (
            min_pow
            and min_pow.value
            and math.isclose(min_pow.value.base_value, 0.0)
        )
```